src/Loader.js

```diff
@@ -280,9 +280,24 @@ export class Loader {
      *     .add('key', 'http://...', { crossOrigin: true }, function () {})
      *     .add('http://...', { crossOrigin: true }, function () {});
      * ```
-     *
-     * @param {string|IAddOptions} [name] - The name of the resource to load, if not passed the url is used.
-     * @param {string} [url] - The url for this resource, relative to the baseUrl of this loader.
+     */
+    /**
+     * @function
+     * @variation 1
+     * @param {...Array<any>} params
```

Comment:
Could these `any`s be `IAddOptions` instead?

Reply:
Doesn't seem to work without `any`s.
Btw, these signatures are based on pixi.d.ts

Reply:
```
add(param: string[] | IAddOptions[]): this;
  add(obj: string | string[] | IAddOptions | IAddOptions[], options?: IAddOptions, cb?: Loader.OnCompleteSignal): this;
  add(name: string, url: string, options?: IAddOptions, cb?: Loader.OnCompleteSignal): this;

```

what about this?
I guess we have to drop rest param,  I this case.

Reply:
I'd prefer to drop the rest param. The overloads I'd like to see are the same as the documentation example above:

```ts
add(key: string, url: string, callback?: Loader.OnCompleteSignal): this;
add(key: string, url: string, options?: IAddOptions, callback?: Loader.OnCompleteSignal): this;
add(url: string, callback?: Loader.OnCompleteSignal): this;
add(url: string, options?: IAddOptions, callback?: Loader.OnCompleteSignal): this;
add(options: IAddOptions, callback?: Loader.OnCompleteSignal): this;
add(resources: (IAddOptions|string)[], callback?: Loader.OnCompleteSignal): this;
```

Reply:
looks good.. I will update the PR